### ISSUE TITLE
Migrate Dart VM to use new common dart front end for both development and deployment of Flutter apps

### ISSUE THREAD

**Issue by a-siva** [dart-lang/sdk#28048]
This is a meta bug to track all open issues that need resolution before we can declare feature complete for the task of migrating the Dart VM to use the new common front end for both development and deployment of flutter apps.

**Comment by a-siva** [comment]
Burn down list
- #28050 Support in flutter tools for new dart front end
- #28051 Support in flutter tools for incremental compilation of sources that have changed
- #28052 Support for incremental kernel files
- #28053 Address current workaround of creating a concatenated bootstrap file
- #28054 Local variable descriptors are not populated currently
- #28055 Support for tokenstreams in Script files
- #28056 Support for starting up from a corelib snapshot when using kernel files

Nice to have list
- #28057 Transform kernel file directly into Dart VM heap objects when reading
-

**Comment by zanderso** [comment]
Adding these from the email thread:

- [ ] #28088 Performance of command line dart running dart programs is acceptable.
- [ ] #28089 Performance of initial load and startup of a flutter app and typical edit/refresh cycle on a flutter app is acceptable.
- [ ] #28090 Performance of hot reloading a flutter app is acceptable.

@a-siva @mhausner @johnmccutchan @sigmundch

**Comment by mhausner** [comment]
A specific bug regarding debugger support:

- [ ] #28570 VM hangs when reaching breakpoint in code loaded from Kernel binary

**Comment by mhausner** [comment]
A specific issue number to track snapshot size of Kernel-based snapshots:

- [ ] Kernel-based VM snapshot size will increase by 4-5x

**Comment by Stargator** [comment]
What do you mean by "Common Front End"? Is there an article or issue where I can get a better understand of it?

**Comment by Stargator** [comment]
Thanks @zoechi

**Comment by mraleph** [comment]
Given that we shipped Dart 2 CFE based pipeline to Flutter this issue should probably be closed. What do you think @a-siva?

**Comment by a-siva** [comment]
There are 2 issues in this list that are still open and can be tracked over there

 #28655 Kernel-based VM snapshot size will increase by 4-5x
 #28088 Performance of command line dart running dart programs is acceptable

Closing this issue.
